oci/casext/gc.go

```diff
@@ -74,11 +81,26 @@ func (e Engine) GC(ctx context.Context) error {
 	}
 
 	n := 0
+sweep:
 	for _, digest := range blobs {
-		if _, ok := black[digest]; ok {
+		desc, ok := black[digest]
+		if ok {
 			// Digest is in the black set.
 			continue
 		}
+
+		// Iterate over all policy funcs and if any one of them indicates that GC
+		// must not be performed, skip that blob
+		for _, policy := range policies {
+			ok, err := policy(desc)
```

Comment:
... when you call `policy` here. So the policy will always see `ispec.Descriptor{}`.